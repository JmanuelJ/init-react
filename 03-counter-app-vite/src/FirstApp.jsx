// const newMessage = {
//     message: 'Hola Mundo',
//     title: 'Juan '
// }

// const getSaludo = (message, title) => {
//     return message + title;
// }


// export const FirstApp = (props) => {
//   return (
//     <>
//     <h1> {getSaludo(newMessage.title, newMessage.message)} </h1>
//     <div>Soy un subtitulo</div>
//     </>
//   )
// }

// export const FirstApp = (props) => {
//   return (
//     <>
//     <h1> {props.title} </h1>
//     <div>Soy un subtitulo</div>
//     </>
//   )
// }

import Proptypes from 'prop-types';

//DesestructuracioÃ³n de props
export const FirstApp = ({
  title,
  subTitle = 'No hay subtitulo',
  name ='JuanMa'
}) => {
  return (
    <>
    <h1 data-testid="test-title">{title}</h1>
    <p>{subTitle}</p>
    <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: Proptypes.string.isRequired,
  subTitle: Proptypes.string
}
