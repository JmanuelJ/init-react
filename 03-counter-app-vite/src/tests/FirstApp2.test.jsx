
import { getAllByTestId, render, screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp";

describe('Pruebas en <FirstApp/>', () => { 
    const title    = 'Hola soy JuanMa';
    const subtitle = 'Soy un subtitulo'

    test('debe de hacer match con el snapshot', () => {
      const {container} = render( <FirstApp title={title}/>);

      expect(container).toMatchSnapshot();
    });

    test('debe mostrar el mensaje "Hola soy JuanMa"', () => { 
      
      render( <FirstApp title={title}/>);
      expect(screen.getByText(title)).toBeTruthy();
    });

    test('debe mostrar el titulo en un h1', () => { 

      render( <FirstApp title={title}/>);

      expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    });

    test('debe de mostrar el subtitulo enviado por props', () => { 
      
      render( 
        <FirstApp 
          title={title}
          subTitle={subtitle}
        />
      );

      expect(screen.getAllByText(subtitle)).toBeTruthy();
     })

 });