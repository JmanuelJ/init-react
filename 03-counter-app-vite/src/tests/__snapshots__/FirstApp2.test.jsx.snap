// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy JuanMa
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    JuanMa
  </p>
</div>
`;
